libraries/resource_provider.rb

```diff
@@ -221,6 +221,18 @@ def get_diff(resource, desired_data)
     def recursive_diff(data, desired_data, str = '', indent = '')
       OneviewCookbook::Helper.recursive_diff(data, desired_data, str, indent)
     end
+
+    # Generic method to retrieve and return a resource from different resources using the type and name of the resource
+    # @param [String, Symbol] resource_class_type Type of resource to be retrieved. e.g., :GoldenImage, :FCNetwork.
+    # @param [String] resource_id Name for this resource. 'Resource1'
+    # @param [String, Symbol] ret_attribute Attribute of resource to be returned by this method. Use :resource for returning complete resource.
+    def load_resource(resource_class_type, resource_id, ret_attribute = :uri)
```

Comment:
I'd probably vote for returning the whole resource by default, but we can easily change that later. I'm working on a different PR that may be able to use it, and I can refactor it then if need-be.